### DISCUSSION TITLE
Relationship between bugs and commits?

### DISCUSSION THREAD

**Discussion by j-xella** [git-bug/git-bug#697] [Q&A]
Hi, is ther a way to tie bugs and commits?  Something like:

1. When I do commit, I mention git-bug ticket in description
2. The commit gets associated with a ticket. That could lead to various possibilites, for example:
   * A new "comment" appears in the ticket, which allows to see that commit details
   * There is a way so see all commits associated with the ticket

**Comment by GlancingMind** [comment]
This is similar to the linked Issues within Pull Requests, but your proposal seems to be a lot more granular.
Unfortunately this could lead into problems. I'll try to make an example upon your given use-case:

> A new "comment" appears in the ticket, which allows to see that commit details

Suppose you have an issue whose solution is composed of 5 commits. Then you have to write down the issue id for each commit (as all of those commits are related to this bug). Leading to multiple commit-comments being present in the bugs timeline. Now suppose some other user proposes also a fix. The bugs timeline would be flooded with implementation specific commit-comments.

---

It looks like, that commits are an implementation detail of a proposed solution to some given problem.
So it might be better to embed those commits within a solution, which then can be linked to an issue.
This proposed solution can then effectively appear once on the bugs timeline.
In other terms, we are back at Pull-Requests/Merge-Requests/Patchset.

Just my 2 cents. Let me know if I misunderstood you. :-)

**Comment by mcepl** [comment]
Could it be something so simple as this `hook/post-commit`:

```sh
#!/bin/sh
set -eux

BUG_ID=$(git format-patch --stdout HEAD~|git interpret-trailers --parse|awk -F '#' '/^Fixes: gb#/ { print $2}')
if [ -n "$BUG_ID" ] ; then
    git bug comment add "${BUG_ID}" -m "Closed by the commit $(git log -1 --abbrev=12 --format="format:%h" HEAD)."
    git bug status close "$BUG_ID"
fi
```

**Reply by mcepl** [reply]
Couldn’t … I need to deal with more `Fixes:` lines, and with bugs which were already closed.

**Reply by mcepl** [reply]
Update for our current weird subcommand structure:

```sh
#!/bin/sh
set -eu

BUG_ID=$(git format-patch --stdout HEAD~|git interpret-trailers --parse\
    |awk -F '#' '/^Fixes: gb#/ { print $2}')
if [ -n "$BUG_ID" ] ; then
    git bug bug comment new "${BUG_ID}" \
        -m "Closed by the commit $(git log -1 --abbrev=12 --format="format:%h" HEAD)."
    git bug bug status close "$BUG_ID"
fi
```

(now at https://git.sr.ht/~mcepl/hlupak/tree/master/item/git-bug-post-commit.sh)
